spec/system/ai_helper/ai_proofreading_spec.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+RSpec.describe "AI Composer Proofreading Features", type: :system, js: true do
+  fab!(:user) { Fabricate(:admin) }
+
+  before do
+    Group.find_by(id: Group::AUTO_GROUPS[:admins]).add(user)
+    assign_fake_provider_to(:ai_helper_model)
+    SiteSetting.ai_helper_enabled = true
+    sign_in(user)
+  end
+
+  let(:composer) { PageObjects::Components::Composer.new }
+
+  it "proofreads selected text using the composer toolbar" do
+    visit "/new-topic"
+    composer.fill_content("hello worldd !")
+
+    # Select a portion of text
+    page.execute_script("document.querySelector('.d-editor-input').setSelectionRange(6, 12);")
+
+    DiscourseAi::Completions::Llm.with_prepared_responses(["world"]) do
+      ai_toolbar = PageObjects::Components::SelectKit.new(".toolbar-popup-menu-options")
+      ai_toolbar.expand
+      ai_toolbar.select_row_by_name("Proofread Text")
+
+      find(".composer-ai-helper-modal .btn-primary.confirm").click
```

Comment:
You can use the generic modal page object for this:

```ruby
let(:modal) { PageObjects::Modals::Base.new }
modal.click_primary_button
```

Reply:
did not seem to work I just left as is for now.